src/background.js

```diff
@@ -1,66 +1,72 @@
 chrome.runtime.onInstalled.addListener(() => {
-  Crawler.setUp();
+  run();
 });
 
 chrome.runtime.onStartup.addListener(() => {
-  Crawler.setUp();
+  run();
 });
 
 
-const Crawler = (() => {
-  var curatedDomains = [];
+function run() {
+  crawler = new Crawler();
+  crawler.setUp();
+}
 
-  return {
-    loadCuratedDomains: () => {
-      const url = chrome.runtime.getURL('../../assets/data/hn-top-domains.json');
-      fetch(url).then(response => {
-        response.json().then(data => {
-          curatedDomains = Object.keys(data);
-          console.log("Loaded curated domains", curatedDomains);
-        });
+
+class Crawler {
+  constructor() {
+    this.curatedDomains = [];
+  }
+
+  loadCuratedDomains() {
+    const url = chrome.runtime.getURL('../../assets/data/hn-top-domains.json');
+    fetch(url).then(response => {
+      response.json().then(data => {
+        this.curatedDomains = Object.keys(data);
+        console.log("Loaded curated domains", this.curatedDomains);
       });
-    },
+    });
+  }
```

Comment:
Here you can make this function `async` and use `await` to make it cleaner and more readable.